packages/accounts-password/password_server.js

```diff
@@ -67,6 +67,20 @@ Accounts._checkPassword = function (user, password) {
 
   if (! bcryptCompare(password, user.services.password.bcrypt)) {
     result.error = handleError("Incorrect password", false);
+  } else if (
+    user.services.password.bcrypt && 
+    Accounts._bcryptRounds > 
+      Number(user.services.password.bcrypt.substring(4, 6))
```

Comment:
Will this work if `bcryptRounds >= 100` (i.e., more than two digits)?

Reply:
While you're at it, let's just extract a helper function called something like `getRoundsFromBcryptHash`?

Reply:
Yeah, that code will break with a work factor above 99, but it shouldn't be possible to implement bcrypt with such a high work factor since it gets expanded geometrically (rounds = 2^[work factor]). [This](http://web.mit.edu/freebsd/head/secure/lib/libcrypt/crypt-blowfish.c) FreeBSD implementation of bcrypt references a maximum of 31 rounds. I'm not sure if 31 is implementation-specific or the theoretical maximum for bcrypt, but the effective maximum for most current server hardware is probably 16 or 17. From benchmarking on my MacBook Pro, hashing with a work factor of 20 takes about 2.4 minutes and 31 would take ~12 years. The rule of thumb I'm aware of (totally unattributed) is that hashing should take ~1 second with current server hardware. If I can find a source for that rule of thumb, I'll do a PR for this in the Guide.

Reply:
Sure, I'm happy to move that into a helper function. At the risk of over-building but with a mind toward implementing other hashing algorithms, I'll grab both the algorithm and whatever parameters go with that algorithm.